### PR TITLE
V2 setup

### PR DESCRIPTION
**Pull request by airportyh**
Why
===

Start conversion to V2 style modules. See [more here](https://docs.google.com/document/d/1Khv0ZNCPaWWma6Oh3Q7P5KMI4JnNPSpejcNVoaEDLYc).

What changed
============

Some preliminary setup for making the changes.

Test plan
=========

`nix eval .#allModulesOptions --json` should give an empty object.

### CODE REVIEW

File: pkgs/default.nix [hunk 1]
```diff
@@ -116,4 +116,63 @@ rec {
...
+    # (import ./modules/formatters/prettier)
+    # (import ./modules/languageServers/typescript-language-server)
+    # (import ./modules/packagers/nodejs-packager)
+    # (import ./modules/debuggers/js-debug)
+    # (import ./modules/bundles/bun)
+    # (import ./modules/interpreters/bun)
+    # (import ./modules/packagers/bun)
+    # (import ./modules/bundles/web)
+    # (import ./modules/languageServers/css-language-server)
+    # (import ./modules/languageServers/html-language-server)
```
Comment (cdmistman) [inline pkgs/default.nix:140]:
```suggestion
```

the dir structure of this repo has really been bothering me, let's fix that with this new era 😅

File: pkgs/default.nix [hunk 2]
```diff
@@ -116,4 +116,63 @@ rec {
...
+
+  myEvalModule = path:
+    (pkgs.lib.evalModules {
+      modules = [
+        (import path)
+      ] ++ allModules;
+      specialArgs = {
+        inherit pkgs pkgs-23_05;
+        pkgs-unstable = pkgs;
+        modulesPath = builtins.toString ./.;
```
Comment (cdmistman) [inline pkgs/default.nix:151]: does this need to be a string? can't we pass the path directly? i know there's no *real* difference between the 2 but it feels better to be passing a path than a string

File: pkgs/default.nix [hunk 3]
```diff
@@ -116,4 +116,63 @@ rec {
...
+      };
+    });
+    lib = pkgs.lib;
+    options = eval.options;
+    filteredOptions = builtins.removeAttrs options ["_module" "description" "displayVersion" "id" "name" "replit"];
+    docsJson = (pkgs.nixosOptionsDoc {
+      options = filteredOptions;
+    }).optionsJSON;
+    in filteredOptions;
```
Comment (cdmistman) [inline pkgs/default.nix:176]: can you `nix run nixpkgs#alejandra -- <this file path>` please?

**Review by cdmistman — COMMENTED** [review]
i'm a bit concerned about this breaking v1 nixmodule dev, are there any breaking changes here? maybe it makes sense to make a new repo?

File: pkgs/default.nix (hunk 1 above)
Comment (airportyh) [reply pkgs/default.nix:140]: What's the suggested change? I can't see the diff...

**Comment by airportyh** [comment]
> i'm a bit concerned about this breaking v1 nixmodule dev, are there any breaking changes here? maybe it makes sense to make a new repo?

We can make it backwards compatible later. "Nixmodules V2 merged to main" is [milestone 2.](https://linear.app/replit/project/nix-modules-v2-5bed040f363e)

File: pkgs/default.nix (hunk 3 above)
Comment (airportyh) [reply pkgs/default.nix:176]: I'll do formatting in a separate pr.

File: pkgs/default.nix (hunk 2 above)
Comment (airportyh) [reply pkgs/default.nix]: Okay. I also changed it to the modules directory. We are not using it anywhere though.

**Comment by airportyh** [comment]
Merging this to get the prototype going. But will address concerns separately.

**Pull request merged**